p2p/p2p.go

```diff
@@ -131,9 +156,49 @@ func (s *Service) Broadcast(msg Message) (err error) {
 	return err
 }
 
-// handles stream; reads message length, message type, and decodes message based on type
-// TODO: implement all message types; send message back to peer when we get a message; gossip for certain message types
-func (s *Service) handleStream(stream net.Stream) {
+// handleP2pStream handles the stream, and rebroadcasts the message based on it's type
+func (s *Service) handleP2pStream(stream net.Stream) {
+
+	msg, err := s.handleStream(stream)
+
+	if err != nil {
+		return
+	}
+
+	// Rebroadcast all messages except for status messages
+	if msg.GetType() != StatusMsgType {
+		err = s.Broadcast(msg)
+		if err != nil {
+			log.Debug("failed to broadcast message: ", err)
+			return
+		}
+	}
+}
+
+var _ module.P2pApi = &Service{}
+
+// ID returns the host's ID
+func (s *Service) ID() string {
+	return s.host.id()
+}
+
+// Peers returns connected peers
+func (s *Service) Peers() []string {
+	return PeerIdToStringArray(s.host.h.Network().Peers())
+}
+
+// PeerCount returns the number of connected peers
+func (s *Service) PeerCount() int {
+	return s.host.peerCount()
+}
+
+// NoBootstrapping returns true if bootstrapping is disabled, otherwise false
+func (s *Service) NoBootstrapping() bool {
+	return s.host.noBootstrap
+}
+
+// handleStream reads message length, message type, decodes message based on type, and returns the decoded message
+func (s *Service) handleStream(stream net.Stream) (Message, error) {
```

Comment:
This doesn't quite achieve what we discussed:
- This is not meant to be another stream handler it should be helper function that parses the message
- Service is the receiver

To rectify this, please:
- Rename this function to `parseMessage`, revert `handleP2pStream` to `handleStream`
- Remove the receiver from the function sig
- Move the write to `msgSendChannel` back up to `handleStream`